File: task_3/task_3.ex
defmodule Solution do
  def solve do
    # Считываем n
    input_n = IO.gets("") |> String.trim()

    if input_n != "" do
      n = String.to_integer(input_n)

      # Считываем числа и считаем квадраты
      kvad = Enum.reduce(1..n, 0, fn _, acc ->
        number = IO.gets("") |> String.trim() |> String.to_integer()

        # Проверка на полный квадрат
        num_sqrt = :math.sqrt(number) |> trunc()

        if num_sqrt * num_sqrt == number do
          acc + 1
        else
          acc
        end
      end)

      IO.puts("#{kvad}")
    end
  end
end

Solution.solve()

File: task_3/task_3.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main(){
    int n;
    int number;
    int kvad = 0; // количество чисел полные квадраты
    cin >> n;
    for (int i = 1; i<=n; i++){
        cin >> number;
        int numSqrt = sqrt(number);
        if ( (numSqrt * numSqrt) == number){
            kvad +=1;
        }
    }
    cout << "Результат: " << kvad <<endl;
    return 0;
}